### DISCUSSION TITLE
Issues while Upgrading from upbound aws provider version 0.35 -> 0.36

### DISCUSSION THREAD

**Discussion by ashishvaishno** [crossplane/crossplane#4183] [Q&A] [answered]
Hello,

I wanted to upgrade to the latest provider version. Just wanted to get some clarity towards the upgrade process with you 😄 
I used the below command to upgrade my provider 
```
kubectl patch provider.pkg crossplane-aws-provider -p '{"spec":{"package": "xpkg.upbound.io/upbound/provider-aws:v0.36.0"}}' --type=merge 
```
On doing so, my providerrevision has the below status 
```
  Warning  SyncPackage        50m                packages/providerrevision.pkg.crossplane.io  cannot establish control of object: policies.acmpca.aws.upbound.io is already controlled by ProviderRevision crossplane-aws-provider-c757062bcb4a (UID a365fc5f-6fba-4e9c-9685-7459cb2ed410)
  Warning  SyncPackage        13m                packages/providerrevision.pkg.crossplane.io  cannot establish control of object: rulegroups.networkfirewall.aws.upbound.io is already controlled by ProviderRevision crossplane-aws-provider-c757062bcb4a (UID a365fc5f-6fba-4e9c-9685-7459cb2ed410)
  Warning  SyncPackage        11m                packages/providerrevision.pkg.crossplane.io  cannot update package revision object metadata: client rate limiter Wait returned an error: context canceled
  Warning  SyncPackage        7m23s              packages/providerrevision.pkg.crossplane.io  cannot establish control of object: Put "https://10.100.0.1:443/apis/apiextensions.k8s.io/v1/customresourcedefinitions/streams.kinesisvideo.aws.upbound.io?dryRun=All": context deadline exceeded
  Warning  SyncPackage        4m20s              packages/providerrevision.pkg.crossplane.io  cannot establish control of object: client rate limiter Wait returned an error: rate: Wait(n=1) would exceed context deadline 
```
* I deleted the crd policies.acmpca.aws.upbound.io and rulegroups.networkfirewall.aws.upbound.io (NOTE : I dont have any managed resources on the cluster yet apart from ClusterAuth)
* I did find a relevant post [here](https://github.com/crossplane/crossplane/issues/2197#issuecomment-796818872), but unfortunately as I dont have any such resources deployed, I was not able to find a solution for me

My setup : 
* I have crossplane installed on a dedicated managed eks cluster (1.26)

**Comment by turkenh** [comment, marked as answer]
Hi @ashishvaishno,

Your upgrade procedure looks fine, but the error messages with `context deadline exceeded` or `context canceled` seems to be an indication of the API server struggling with handling a high number of CRDs. During upgrades, it causes more load on the API than installation since the old `ProviderRevision` is trying to update old CRDs to stop controlling them, and the new `ProviderRevision` is trying to apply new CRDs while trying to gain their controller at the same time. 

Did you already consider [migrating](https://docs.upbound.io/knowledge-base/migrate-to-provider-families/) to [provider families](https://blog.crossplane.io/crd-scaling-provider-families)?

**Comment by jbw976** [comment]
It looks like @turkenh's assessment here is accurate, where there are probably too many resources in the control plane for the upgrade to be resolved smoothly without errors.

As we have rolled out the provider families this week that specifically intends to resolve this type of issue, I think there's nothing specific left to solve in Crossplane at this time.  I'm going to convert this issue to a discussion so that it's still visible and the community can chat/help/support more :)

**Comment by ashishvaishno** [comment]
@jbw976 @turkenh and the whole upbound team/community thanks for releasing the feature 🕺
